app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -762,6 +762,45 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  fun testStateFragment_nextState_viewHint_configChange_noNewHintAvailable() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+      produceAndViewFirstHint()
+
+      submitWrongAnswerToState2()
+      onView(isRoot()).perform(orientationLandscape())
```

Comment:
I'm actually wondering whether this will correctly repro the issue. I believe @FareesHussain had another way to solve the problem but couldn't actually write tests for it due to Robolectric limitations. Curious to see per the above if these tests actually pass/fail as expected.

Reply:
This approach solves the two existing issues that were present in the hint hander
1. Hints that were scheduled to be visible when the timer runs out, are visible even on the completed state.
2. new hints were revealed (after the first hint was viewed) upon configuration change.

I have verified that without this fix the tests that check the above two cases fail (added screenshot in pr description).

All tests pass for the current approach.